compiler/src/codegen/compcore.re

```diff
@@ -1649,14 +1646,14 @@ let allocate_string = (wasm_mod, env, str) => {
       ),
       Expression.Const.make(
         wasm_mod,
-        const_int32(tag_val_of_heap_tag_type(StringType)),
+        const_int32(tag_val_of_heap_tag_type(tag)),
       ),
     ),
     store(
       ~offset=4,
       wasm_mod,
       get_swap(),
-      Expression.Const.make(wasm_mod, const_int32 @@ String.length(str)),
+      Expression.Const.make(wasm_mod, const_int32 @@ len),
```

Comment:
Any reason to do this over `Buffer.length`?

Reply:
I wasn't sure if `Bytes.length` and `String.length` would yield the same value. I will change that!